src/js/services/app.js

```diff
@@ -74,24 +75,41 @@ export function showChannelPage(channelType) {
         const {user, app: {integrations}} = getState();
         const channelDetails = CHANNEL_DETAILS[channelType];
         const isLinked = isChannelLinked(user.clients, channelType);
-        const openLink = channelDetails.getURL && (!channelDetails.Component || isLinked);
 
-        if (openLink) {
+        function _showChannelPage() {
+            dispatch(AppStateActions.showChannelPage(channelType));
+
+            return dispatch(connectToFayeUser())
+                .then(() => dispatch(channelDetails.onChannelPage()));
+        }
+
+        function openLink() {
             const appChannel = getIntegration(integrations, channelType);
-            const link = channelDetails.getURL(user, appChannel, isLinked);
+            const getCode = channelDetails.usesCode ?
```

Comment:
I didn't implement any UI feedback, it didn't seem necessary since it's a pretty fast API.

Wasn't aware of the popup blocker danger. Assuming the code was safe before, do you know how popup blockers were linking the UI event to the window.open before this change? Do you know much of a delay might cause popup blocker to intervene?